### PR TITLE
[ch3109] Have an option to not display an category and all articles linked

### PR DESCRIPTION
**Pull request by cacahouwete**
https://app.clubhouse.io/enmarche/story/3109/etq-qu-editor-je-veux-creer-une-categorie-d-article-qui-ne-s-affiche-pas-sur-la-page-d-actualit%C3%A9s-et-dont-les-articles-ne

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -84,6 +85,8 @@ db: vendor wait-for-db
 db-diff:        ## Generate a migration by comparing your current database to your mapping information
 db-diff: vendor wait-for-db
 	$(EXEC) $(CONSOLE) doctrine:migration:diff
+	sudo chown -R ${whoami}:${whoami} app/migrations
```
[the remark this answers is not in the document]
Comment (B-Galati) [reply Makefile]: Please don't create a new variable for the moment. It's not needed.

File: src/DataFixtures/ORM/LoadArticleData.php [hunk 2]
```diff
@@ -91,8 +92,21 @@ public function load(ObjectManager $manager)
             'amp_content' => file_get_contents(__DIR__.'/../content_amp.html'),
         ]));
 
+        $manager->persist($factory->createFromArray([
+            'title' => 'Article dans une category pas display',
```
Comment (thomasbisignani) [inline src/DataFixtures/ORM/LoadArticleData.php]: Please don't mixin french and english to be more consistant.

File: src/DataFixtures/ORM/LoadArticleData.php [hunk 3]
```diff
@@ -91,8 +92,21 @@ public function load(ObjectManager $manager)
...
+            'slug' => 'article-with-category-no-display',
+            'description' => 'Article dans une category pas display',
```
Comment (thomasbisignani) [inline src/DataFixtures/ORM/LoadArticleData.php]: Same here

File: src/DataFixtures/ORM/LoadArticleData.php [hunk 4]
```diff
@@ -91,8 +92,21 @@ public function load(ObjectManager $manager)
             'amp_content' => file_get_contents(__DIR__.'/../content_amp.html'),
         ]));
 
+        $manager->persist($factory->createFromArray([
+            'title' => 'Article dans une category pas affiché',
+            'slug' => 'article-avec-category-non-afficher',
+            'description' => 'Article dans une category pas affiché',
```
Comment (thomasbisignani) [inline src/DataFixtures/ORM/LoadArticleData.php]: ` 'description' => 'Article dans une categorie pas affichée',`

File: src/DataFixtures/ORM/LoadArticleData.php [hunk 5]
```diff
@@ -91,8 +92,21 @@ public function load(ObjectManager $manager)
             'amp_content' => file_get_contents(__DIR__.'/../content_amp.html'),
         ]));
 
+        $manager->persist($factory->createFromArray([
+            'title' => 'Article dans une category pas affiché',
```
Comment (thomasbisignani) [inline src/DataFixtures/ORM/LoadArticleData.php]: ` 'description' => 'Article dans une catégorie pas affichée',`

File: src/Admin/ArticleAdmin.php [hunk 6]
```diff
@@ -108,6 +108,9 @@ protected function configureDatagridFilters(DatagridMapper $datagridMapper)
                 'label' => 'Titre',
                 'show_filter' => true,
             ])
+            ->add('category', null, [
+                'label' => 'Category',
```
Comment (ottaviano) [inline src/Admin/ArticleAdmin.php]: in FR pls

File: src/Repository/ArticleRepository.php [hunk 7]
```diff
@@ -46,7 +41,9 @@ public function findByCategoryPaginated(string $category, int $page, int $perPag
         $qb = $this->createQueryBuilder('a')
             ->select('a', 'm')
             ->leftJoin('a.media', 'm')
+            ->leftJoin('a.category', 'category')
```
Comment (ottaviano) [inline src/Repository/ArticleRepository.php]: why left join and not inner?

Comment (cacahouwete) [reply src/Repository/ArticleRepository.php]: good question , I think it s better, but why no inner join for 

```

            $qb
                ->addSelect('c')
                ->leftJoin('a.category', 'c')
                ->andWhere('c.slug = :category')
                ->setParameter('category', $category)
            ;
```

?

File: src/Admin/ArticleCategoryAdmin.php [hunk 8]
```diff
@@ -30,6 +30,9 @@ protected function configureFormFields(FormMapper $formMapper)
             ->add('ctaLabel', null, [
                 'label' => 'Label d\'action',
             ])
+            ->add('display', null, [
+                'label' => 'Affiché ?',
```
Comment (Remg) [inline src/Admin/ArticleCategoryAdmin.php]: Affiché**e**

File: src/Admin/ArticleCategoryAdmin.php [hunk 9]
```diff
@@ -51,6 +54,10 @@ protected function configureListFields(ListMapper $listMapper)
             ->add('ctaLabel', null, [
                 'label' => 'CTA Label',
             ])
+            ->add('display', null, [
+                'label' => 'Affiché ?',
```
Comment (Remg) [inline src/Admin/ArticleCategoryAdmin.php]: Affiché**e**

File: features/sitemap.feature [hunk 10]
```diff
@@ -33,8 +33,9 @@ Feature:
 
   Scenario: The content sitemap displays correctly
     Given I am on "/sitemap_content_1.xml"
-    Then the response should be in XML
-    And the XML element "//urlset" should have 183 element
+    Then the response status code should be 200
+    And the response should be in XML
+    And the XML element "//urlset" should have 210 element
```
[the remark this answers is not in the document]
Comment (cacahouwete) [reply features/sitemap.feature]: Yes ^^, it's weird but that's what marine told me ^^, juste not display in /artilces category and articles , but all the other yes

[the remark this answers is not in the document]
Comment (illusionOfParadise) [reply features/sitemap.feature]: @cacahouwete Did you define this also with Julien?

File: tests/Command/AlgoliaSynchronizeCommandTest.php [hunk 11]
```diff
@@ -40,7 +40,7 @@ public function dataProviderTestCommand(): array
             [
                 [], // no parameters
                 [
-                    'Synchronizing entity AppBundle\Entity\Article ... done, 154 records indexed',
+                    'Synchronizing entity AppBundle\Entity\Article ... done, 180 records indexed',
```
[the remark this answers is not in the document]
Comment (cacahouwete) [reply tests/Command/AlgoliaSynchronizeCommandTest.php]: It's the same response "it's weird but that's what marine told me ^^, juste not display in /artilces category and articles , but all the other yes"

File: src/Repository/ArticleRepository.php (hunk 7 above)
Comment (cacahouwete) [reply src/Repository/ArticleRepository.php]: I will transform theses things with inner join ^^, it's more logical

File: features/sitemap.feature (hunk 10 above)
[the remark this answers is not in the document]
Comment (marineb) [reply features/sitemap.feature]: @illusionOfParadise no we just hide them from the article feed. and we hide that category from the article page but we want them to be in the sitemap, and to be indexed.

File: tests/Command/AlgoliaSynchronizeCommandTest.php (hunk 11 above)
[the remark this answers is not in the document]
Comment (marineb) [reply tests/Command/AlgoliaSynchronizeCommandTest.php]: We do want them to appear on search results, to be indexed, we just don\t want them to pollute the article landing page feed.

File: features/sitemap.feature (hunk 10 above)
[the remark this answers is not in the document]
Comment (illusionOfParadise) [reply features/sitemap.feature]: @marineb ok, thanks.

File: src/Repository/ArticleRepository.php [hunk 12]
```diff
@@ -46,18 +41,20 @@ public function findByCategoryPaginated(string $category, int $page, int $perPag
...
             ->setMaxResults($perPage)
             ->setFirstResult(($page - 1) * $perPage)
         ;
 
         if (!ArticleCategory::isDefault($category)) {
             $qb
-                ->addSelect('c')
-                ->leftJoin('a.category', 'c')
-                ->andWhere('c.slug = :category')
+                ->addSelect('category')
```
[the remark this answers is not in the document]
Comment (cacahouwete) [reply src/Repository/ArticleRepository.php]: I just change the naming, I don't think it's a good idea to move it in main query

[the remark this answers is not in the document]
Comment (ottaviano) [reply src/Repository/ArticleRepository.php]: :D naming? 
you change more that juste naming, you remove the join here, and add the inner join to main query.

**Pull request merged**